torch/_dynamo/utils.py

```diff
@@ -1645,6 +1645,8 @@ def to_numpy_helper(value):
 
 def numpy_to_tensor(value):
     """Convert tnp.ndarray to tensor, leave other types intact. If a list/tuple, loop through it to convert."""
+    if isinstance(value, np.ndarray):
+        return numpy_to_tensor(tnp.array(value, dtype=str(value.dtype)))
```

Comment:
Let's save a few roundtrips
```suggestion
        return torch.as_tensor(value)
```

Reply:
Can only do as_tensor if you get an ndarray. For other types you need different handling.

Reply:
Sg